src/features/studio/ActionBar.tsx

```diff
@@ -0,0 +1,204 @@
+import { ActionBar } from 'src/components';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useSigningClient } from 'src/contexts/signerClient';
+import { useBackend } from 'src/contexts/backend/backend';
+import useWaitForTransaction from 'src/hooks/useWaitForTransaction';
+import { addIfpsMessageOrCid } from 'src/utils/ipfs/helpers';
+import { InputMemo } from 'src/pages/teleport/components/Inputs';
+import { PATTERN_IPFS_HASH } from 'src/constants/patterns';
+import { sendCyberlinkArray } from 'src/services/neuron/neuronApi';
+import { KeywordsItem, useStudioContext } from './studio.context';
+import { useAdviser } from '../adviser/context';
+import { AdviserColors } from '../adviser/Adviser/Adviser';
+import { checkLoopLinks, mapLinks, reduceLoopKeywords } from './utils/utils';
+
+function ActionBarContainer() {
+  const { signer, signingClient } = useSigningClient();
+  const { isIpfsInitialized, ipfsApi } = useBackend();
+  const { setAdviser } = useAdviser();
+
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string>();
+
+  const [newKeywords, setNewKeywords] = useState<
+    | {
+        value: string;
+        fileName?: string;
+      }
+    | undefined
+  >();
+
+  const {
+    currentMarkdown,
+    keywordsFrom,
+    keywordsTo,
+    setStateActionBar,
+    stateActionBar,
+    addKeywords,
+  } = useStudioContext();
+
+  const [tx, setTx] = useState({
```

Comment:
action bar has a lot of reusable logic, need to clean it, now is complex.
I created new hook, lets do similar. can discuss in element
https://github.com/cybercongress/cyb-ts/blob/master/src/features/cyberlinks/hooks/useCyberlink.ts